index.js

```diff
@@ -92,7 +92,14 @@ function getFirstHeader (req, header) {
 
 function parsePartialURL (url) {
   const containsProtocol = url.indexOf('://') !== -1
-  const result = parseUrl(containsProtocol ? url : 'invalid://' + url)
-  if (!containsProtocol) result.protocol = ''
+  const urlInstance = new URL(containsProtocol ? url : 'invalid://' + url)
```

Comment:
This could *throw* where `url.parse` would not throw before, e.g.:

```
> new URL('invalid://@@')
Uncaught TypeError [ERR_INVALID_URL]: Invalid URL
    at __node_internal_captureLargerStackTrace (node:internal/errors:478:5)
    at new NodeError (node:internal/errors:387:5)
    at URL.onParseError (node:internal/url:565:9)
    at new URL (node:internal/url:641:5) {
  input: 'invalid://@@',
  code: 'ERR_INVALID_URL'
}
> url.parse('invalid://@@')
Url {
  protocol: 'invalid:',
  slashes: true,
  auth: '@',
  host: '',
  port: null,
  hostname: '',
  hash: null,
  search: null,
  query: null,
  pathname: null,
  path: null,
  href: 'invalid://'
}
```

Reply:
Good catch!
I think the legacy `parse` method is much more permissive since it consists on a loop trying to figure out as much as possible from the input string. You can look at [the implementation](https://github.com/nodejs/node/blob/3abbc3829aeb7844eec8ccbfe2cc05cf5fcc4982/lib/url.js#L181)

This means there will be more cases where the [WHATWG URL API](https://nodejs.org/api/url.html#the-whatwg-url-api) would simply throw since the input  does not conform to its specs.

I guess the best approach to do not drop support is to catch the error and fallback to the legacy so we ensure support to package consumers which make use of this edge cases.